app/src/main/java/com/mashup/lastgarden/ui/main/PerfumeInformationFragment.kt

```diff
@@ -25,41 +34,76 @@ class PerfumeInformationFragment : BaseViewModelFragment() {
 
     override fun onSetupViews(view: View) {
         super.onSetupViews(view)
-        setNoteButtonChecked()
+        addListenersOnCheckButton()
     }
 
-    private fun setNoteButtonChecked() {
-        binding.perfumePyramidInclude.topCheckBox.setOnClickListener {
-            allButtonUnchecked()
-            binding.perfumePyramidInclude.apply {
-                topCheckBox.isChecked = true
-                pyramidImageView.setImageResource(R.drawable.ic_pyramid_top)
-                pyramidContentTextView.text = "시트러스 어코드, 페티그레인, 블랙커런트, 레드프룻, 피치"
-            }
+    override fun onBindViewModelsOnCreate() {
+        lifecycleScope.launchWhenCreated {
+            viewModel.perfumeDetailItem
+                .filterNotNull()
+                .collectLatest {
+                    setMainAccords()
+                    setPerfumePyramidVisibility()
+                }
+        }
+        lifecycleScope.launchWhenCreated {
+            viewModel.selectedNote
+                .filterNotNull()
+                .collectLatest { selectedNote ->
+                    binding.perfumePyramidInclude.topCheckBox.isChecked =
+                        selectedNote == PerfumeDetailViewModel.PerfumeDetailNote.TOP
+                    binding.perfumePyramidInclude.middleCheckBox.isChecked =
+                        selectedNote == PerfumeDetailViewModel.PerfumeDetailNote.MIDDLE
+                    binding.perfumePyramidInclude.baseCheckBox.isChecked =
+                        selectedNote == PerfumeDetailViewModel.PerfumeDetailNote.BASE
+                    viewModel.setNoteContents()
+                }
         }
-        binding.perfumePyramidInclude.middleCheckBox.setOnClickListener {
-            allButtonUnchecked()
+        lifecycleScope.launchWhenCreated {
+            viewModel.noteContents
+                .filterNotNull()
+                .collectLatest { noteContents ->
+                    if (viewModel.selectedNote.value != null) {
+                        binding.perfumePyramidInclude.pyramidContentTextView.text = noteContents
+                    } else {
+                        binding.perfumePyramidInclude.noteContentTextView.text = noteContents
+                    }
+                }
+        }
+    }
+
+    private fun setMainAccords() {
+        binding.mainAccordsInclude.accordContentTextView.text = viewModel.mainAccords.value
+    }
+
+    private fun setPerfumePyramidVisibility() {
+        if (viewModel.selectedNote.value == null) {
```

Comment:
ViewModel의 Flow로 바로 접근하는 것 보다 public get 으로 접근 가능한 변수를 따로 만들어서 접근하는게 좋아보입니다!

Reply:
```
private val _selectedNote = MutableStateFlow<PerfumeDetailNote?>(null)
val selectedNote: StateFlow<PerfumeDetailNote?> = _selectedNote
```

ViewModel에서 selectedNote 변수를 public으로 선언하고 view에서 사용하고 있었는데, 혹시 이게 아닐까요..?!

Reply:
Fragment가 selectedNote의 value에 접근하기 위해 StateFlow.value를 하는 것 보다
```suggestion
    val selectedNote: PerfumeDetailNote? get()= _selectedNote.value
```
이렇게 바로 쓸 수 있게 하는게 좋아요! 딱 필요한 scope 까지만 접근하게!
지금은 StateFlow라 상관없지만 만약 저게 MutableStateFlow라면 Fragment에서 data를 offer 할 수 있어서 최대한 그 범위를 줄여주는게 좋습니당

Reply:
우왓 감사합니다!!! 참고해서 수정해보겠습니다!! 🤩🤩👍